File: 프로그래머스/lv0/120864. 숨어있는 숫자의 덧셈 （2）/숨어있는 숫자의 덧셈 （2）.js
function solution(my_string) {
   let sum = 0;
    let currNum = 0;

    for (let i = 0; i < my_string.length; i++) {
        const char = my_string[i];

        if (char >= "0" && char <= "9") {
            currNum = currNum * 10 + parseInt(char);
        } else {
            sum += currNum;
            currNum = 0;
        }
    }
    sum += currNum;
    return sum;
}

File: 프로그래머스/lv0/120864. 숨어있는 숫자의 덧셈 （2）/숨어있는 숫자의 덧셈 （2）.java
import java.util.ArrayList;
import java.util.List;
class Solution {
    public int solution(String my_string) {
        List< Integer > numbers = new ArrayList<>();
		int temp = 0;
		for ( int i = 0; i < my_string.length(); i++ ) {
			char ch = my_string.charAt( i );
			if ( Character.isDigit( ch ) ) {
				temp = temp * 10 + Character.getNumericValue( ch );
			} else {
				numbers.add( temp );
				temp = 0;
			}
		}
        numbers.add( temp );
		return numbers.stream().reduce( 0, Integer :: sum );
    }
}